playbooks/prepare_stack_testconfig.yaml

```diff
@@ -0,0 +1,92 @@
+---
+- hosts: standalone
+  become: true
+  become_user: stack
+  gather_facts: false
+  vars_files: vars/defaults.yaml
+
+  tasks:
+
+  - name: Gather host environment variables
+    setup:
+      filter: ansible_env
+
+  - name: Ensure remote .ssh directory
+    file:
+      path: "{{ ansible_env.HOME }}/.ssh"
+      state: directory
+      owner: stack
+      group: stack
+      mode: '0700'
+
+  - name: Copy ssh public key to host
+    copy:
+      src: "{{ testconfig_public_key }}"
+      dest: "{{ ansible_env.HOME }}/.ssh/{{ testconfig_public_key | basename }}"
+      owner: stack
+      group: stack
+      mode: '0600'
+    register: testconfig_public_key_copy
+
+  - name: Configure resources for openshift user
+    shell: |
+      set -e -o pipefail
+
+      if ! openstack keypair show default >/dev/null; then
+          openstack keypair create --public-key {{ testconfig_public_key_copy.path }} default
+      fi
+      if ! openstack network show private >/dev/null; then
+          openstack network create --internal private
```

Comment:
you can use a flag on the openstck commands to modified their output to lower traffic back and forth. some cli commands result in several rest calls to the openstack api , for example you coud add something like   -f value -c id, 
and theonly thing you should get is the uuid of the object that was created. 

I guess it not much, but i just hate that dam pause when executing an openstack cli.